Int overflow on i686 and armv7hl

When building on a 32 bits x86 or 32 bits ARM system, the following error occurs:

```
Testing: "/builddir/build/BUILD/cachecontrol-fd225a25f1983d116db58553068677ba03f51fa9/_build/src/github.com/pquerna/cachecontrol/cacheobject"
+ GOPATH=/builddir/build/BUILD/cachecontrol-fd225a25f1983d116db58553068677ba03f51fa9/_build:/usr/share/gocode
+ go test -buildmode pie -compiler gc -ldflags '-extldflags '\''-Wl,-z,relro  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld '\'''
# github.com/pquerna/cachecontrol/cacheobject
./directive_test.go:283:43: constant 9223372036854775807 overflows int
FAIL	github.com/pquerna/cachecontrol/cacheobject [build failed]
```
(I'm packaging for Fedora 28 and above. You can see the full build log at https://koji.fedoraproject.org/koji/taskinfo?taskID=27014801)

--- a/cacheobject/directive_test.go
+++ b/cacheobject/directive_test.go
@@ -280,7 +280,7 @@ func TestParseDeltaSecondsLarge(t *testing.T) {
 }
 
 func TestParseDeltaSecondsVeryLarge(t *testing.T) {
-	ds, err := parseDeltaSeconds(fmt.Sprintf("%d", math.MaxInt64))
+	ds, err := parseDeltaSeconds(fmt.Sprintf("%d", int64(math.MaxInt64)))
 	require.NoError(t, err)
 	require.Equal(t, ds, DeltaSeconds(math.MaxInt32))
 }

